ARSIS-Telemetry-GroundControl/LMCC/src/lib/datastore.ts

```diff
@@ -0,0 +1,52 @@
+import { get, writable } from 'svelte/store';
+
+/**
+ * Creates a Svelte store that connects to a WebSocket and listens for JSON messages.
+ * @param {string} url - The WebSocket URL to connect to.
+ * @returns A Svelte store with the WebSocket's messages.
+ */
+export function createWebSocketStore(url: string) {
+	const store = writable(null);
+
+	const ws = new WebSocket(url);
+
+	ws.onopen = () => {
+		console.log('WebSocket connection established');
+	};
+
+	ws.onerror = (error) => {
+		console.error('WebSocket error:', error);
+	};
+
+	ws.onmessage = (event) => {
+		try {
+			const data = JSON.parse(event.data);
+			let oldStore = get(store)
+			let upsert = oldStore ? oldStore.upsert : null;
+			let lastlist = oldStore ? oldStore[data.type] : [];
+			lastlist = lastlist ? lastlist : [];
+			store.set({ ...oldStore, ...{ [data.type]: {...lastlist, ...{ [(upsert ? upsert : data.time)]: Object.values(data.data) }}}})
```

Comment:
Could use nullish coalescing operator instead of ternary:
`upsert ?? data.time` 
instead of 
`upsert ? upsert : data.time`